compaction/compaction_strategy.cc

```diff
@@ -87,17 +87,89 @@ std::optional<sstring> compaction_strategy_impl::get_value(const std::map<sstrin
     return it->second;
 }
 
-compaction_strategy_impl::compaction_strategy_impl(const std::map<sstring, sstring>& options) {
-    using namespace cql3::statements;
+void compaction_strategy_impl::validate_min_max_threshold(const std::map<sstring, sstring>& options, std::map<sstring, sstring>& unchecked_options) {
+    auto min_threshold_key = "min_threshold", max_threshold_key = "max_threshold";
+    long default_min_threshold = 4, default_max_threshold = 32;
+
+    auto tmp_value = compaction_strategy_impl::get_value(options, min_threshold_key);
+    auto min_threshold = cql3::statements::property_definitions::to_long(min_threshold_key, tmp_value, default_min_threshold);
+    if (min_threshold <= 2) {
```

Comment:
2 is a valid min threshold, which means you want to restrict each LSM-tree level to a single sstable.

Reply:
we even use it implicitly on scylla.